dmoj/graders/bridged.py

```diff
@@ -0,0 +1,151 @@
+        # of failing if there's not a lot of swap space available.
+        #
+        # We don't really have a way to force the generated data to disappear, so calling a gc here is the best
+        # chance we have.
+        if platform.python_implementation() == 'PyPy':
+            gc.collect()
+
+        return result
+
+    def check_result(self, case, result):
+        if result.result_flag:
+            # This is usually because of a TLE verdict caused by printing stuff after the interactor
+            # has issued the AC verdict
+            # This results in a TLE verdict getting full points, which should not be the case
+            return CheckerResult(False, 0)
+
+        # We use the testlib.h return codes
+        AC = 0
+        WA = 1
+        PE = 2
+        IE = 3
+
+        interactor = self.interactor
+        interactor_binary = self.interactor_binary
+        show_feedback = self.handler_data.feedback
+        stderr = interactor.stderr.read()
+        point_value = case.points
+
+        if interactor.tle:
+            time_limit = (self.handler_data.preprocessing_time or 0) + self.problem.time_limit
+            raise InternalError('interactor timed out (>%d seconds)' % time_limit)
+        if interactor.mle:
+            memory_limit = self.handler_data.memory_limit or env['generator_memory_limit']
+            raise InternalError('interactor ran out of memory (> %s Kb)' % memory_limit)
+        if interactor.protection_fault:
+            syscall, callname, args = interactor.protection_fault
+            raise InternalError('interactor invoked disallowed syscall %s (%s)' % (syscall, callname))
+
+        if interactor.returncode == AC:
+            if show_feedback:
+                return CheckerResult(True, point_value, feedback=stderr)
+            else:
+                return CheckerResult(True, point_value)
+        elif interactor.returncode in (WA, PE):
+            if show_feedback:
+                return CheckerResult(False, 0, feedback=stderr)
+            else:
+                return CheckerResult(False, 0, feedback='Presentation Error' if interactor.returncode == PE else '')
+        else:
+            if interactor.returncode == IE:
+                error = 'checker failed assertion with message %s' % stderr
+            else:
+                error = 'checker returned unexpected return code %d' % interactor.returncode
+                # To get the feedback, we need a Result object, but we lack a Case object
+                # So we set it to None because we don't need to access it
+                result = Result(None)
+                result.set_result_flag(interactor)
+                if hasattr(interactor_binary, 'feedback') and interactor.feedback:
+                    feedback = interactor.feedback
+                else:
+                    feedback = getattr(interactor_binary, 'get_feedback', lambda x, y, z: '')(stderr, result, interactor)
```

Comment:
I have no idea what this chunk of code does, but I've seen it repeated many times... it shouldn't be